### DISCUSSION TITLE
HTML Minification

### DISCUSSION THREAD

**Discussion by obj-obj** [withastro/roadmap#165] [Proposal]
Add an option to minify HTML when building (remove whitespace)

**Comment by iojcde** [comment]
Has anyone figured out a solution?

**Reply by obj-obj** [reply]
You can install [minhtml](https://www.npmjs.com/package/minhtml) and change your `package.json`'s build step to `astro build && minhtml --collapse-boolean-attributes --collapse-whitespace --decode-entities --remove-comments --remove-empty-attributes --remove-empty-elements --remove-optional-tags --remove-redundant-attributes --remove-script-type-attributes --remove-style-link-type-attributes --use-short-doctype --minify-css --minify-js --file-ext html --input-dir dist --output-dir dist`

**Comment by sondr3** [comment]
I was really surprised to find that Astro didn't do any HTML minification, so I created [`astro-html-minifier`](https://github.com/sondr3/astro-html-minifier) to fix the problem temporarily. I might be interested in upstreaming something like this if there is interest for it/I gain more experience with Astro as I don't know how this interacts with SSR/SSG.

**Comment by patdx** [comment]
I was surprised it's not included too. It seems like the using an external HTML minifier would be a great solution for static build. However it would  probably have some performance and bundle size cost to include that in SSR mode? Especially for Cloudflare and similar edge platforms. Personally I would love to see Astro adopt JSX style whitespace rules, which I think would solve the issue for both static and SSR, but understand that may not be philosophically aligned with Astro.

**Comment by natemoo-re** [comment]
Woohoo! 🎉 This feature has landed on our Roadmap in a Stage 2 proposal. https://github.com/withastro/roadmap/issues/537

Thanks for your patience with this, everyone. I know this feature has been a long time coming—we're looking forward to having it land as a default optimization in `astro`.
